### ISSUE TITLE
trigger assert in to_unsigned

### ISSUE THREAD

**Issue by pauldreik** [fmtlib/fmt#1136]
I am not sure if this is the same as https://github.com/fmtlib/fmt/pull/1135 but it does not trigger for all values inserted into "value" (I tried with -1,0,1).
```cpp
TEST(ChronoTest, Invalid2) {
	const std::chrono::duration<long, std::atto> value(-1787876068541059024L);

    EXPECT_THROW(fmt::format("{:%S", value),
                 fmt::format_error);
}
```
The resulting assert() is triggered in to_unsigned.

**Comment by pauldreik** [comment]
I found a related error, which triggers another assertion but which when minimized (by afl) triggers the same assertion as this. I pushed it onto https://github.com/pauldreik/fmt/blob/fuzz/fuzzing/crashes/chrono_duration/id:000039%2Csig:06%2Csrc:000757%2Cop:flip4%2Cpos:3
To recreate the crash, follow the instructions on https://github.com/pauldreik/fmt/tree/fuzz/fuzzing#reproduce-with-a-normal-build

**Comment by vitaut** [comment]
Thanks for reporting. I think it is related to my quick fix to #1135.

**Comment by vitaut** [comment]
Fixed in https://github.com/fmtlib/fmt/commit/4a4d72f9177aec7a1f2025cf0c5eeda48d932728.
